client/src/app/components/registration/registration.component.ts

```diff
@@ -42,16 +44,19 @@ export class RegistrationComponent {
   }
 
   async submitForm() {
-    //TODO need to update and connect with server endpoint
-    console.log('Registered successfully');
-    console.log(this.registrationForm.value);
+    if (this.registrationForm.valid) {
+      this.userService.createUser(this.registrationForm.value).subscribe(() => {
+        this.router.navigate(['/login']);
+      });
```

Comment:
Nice handling of the observable! But possibly consider handling errors? Otherwise if there's a user error or a bug it's hard to tell what happened:
```
this.userService.createUser(this.registrationForm.value).subscribe({
  next: () => this.router.navigate(['/login']),
  error: (err) => alert('Error signing up: ', err.error.message),
});
```